### PR TITLE
srtp: reuse openssl context for aes encryption

### PR DESCRIPTION
**Pull request by algesten**

### CODE REVIEW

**Review by k0nserv — APPROVED** [review]

**Review by davibe — APPROVED** [review]

File: src/rtp/srtp.rs [hunk 1]
```diff
@@ -358,41 +361,58 @@ impl Derived {
...
-        let mode = if encrypt {
-            Mode::Encrypt
-        } else {
-            Mode::Decrypt
-        };
+impl Encrypter {
+    fn new(key: AesKey) -> Self {
+        let t = cipher::Cipher::aes_128_ctr();
+        let mut ctx = CipherCtx::new().unwrap();
+        ctx.encrypt_init(Some(t), Some(&key[..]), None).unwrap();
```
Comment (xnorpx) [inline src/rtp/srtp.rs:389]: maybe expect instead? (my bad unwraps)

Comment (algesten) [reply src/rtp/srtp.rs:389]: good point!

**Comment by xnorpx** [comment]
LGTM

**Comment by xnorpx** [comment]
There is inplace API, but that can be done later if ever needed.

**Pull request merged**